test/unit/node/util.test.ts

```diff
@@ -189,6 +189,17 @@ describe("isHashMatch", () => {
     const actual = await util.isHashMatch(password, _hash)
     expect(actual).toBe(false)
   })
+  it("should return false and not throw an error if the hash doesn't start with a $", async () => {
+    const password = "hellowpasssword"
+    const _hash = "n2i$v=19$m=4096,t=3,p=1$EAoczTxVki21JDfIZpTUxg$rkXgyrW4RDGoDYrxBFD4H2DlSMEhP4h+Api1hXnGnFY"
+    expect(async () => await util.isHashMatch(password, _hash)).not.toThrow()
+    expect(await util.isHashMatch(password, _hash)).toBe(false)
+  })
+  it("should reject the promise and throw if error", async () => {
+    const password = "hellowpasssword"
```

Comment:
Just curious - what does it mean when the hash begins with a `$`? Is it used as a signifier to designate that the hash is Argon2? I had thought that the other test case was an Argon2 hash, too, so this is a little confusing for me

Reply:
> Just curious - what does it mean when the hash begins with a $? Is it used as a signifier to designate that the hash is Argon2?

AFAIK, I believe that's exactly why. 

>  had thought that the other test case was an Argon2 hash, too, so this is a little confusing for me

If you point out which test case you're referring to, happening to clear up any confusion!

Reply:
Sorry I meant that this:

```
    const _hash = "n2i$v=19$m=4096,t=3,p=1$EAoczTxVki21JDfIZpTUxg$rkXgyrW4RDGoDYrxBFD4H2DlSMEhP4h+Api1hXnGnFY"
```

Looks like a valid hash to me, but I guess it's not? What is this hash?

I guess it makes sense that it has to start with `$` since a valid hash looks like `    const _hash = "$argon2i$v=19$m=4096,t=3,p=1$EAoczTxVki21JDfIZpTUxg$rkXgyrW4RDGoDYrxBFD4H2DlSMEhP4h+Api1hXnGnFY"`?

Should we look for "`$argon2i$`" instead of just the `$`?